src/Symfony/Component/Mailer/Tests/MailerTest.php

```diff
@@ -69,4 +69,16 @@ public function dispatch($message, array $stamps = []): Envelope
         self::assertCount(1, $bus->messages);
         self::assertSame($email, $bus->messages[0]->getMessage());
     }
+
+    public function testCannotSendDraftEmails()
+    {
+        $transport = new Mailer($this->createMock(TransportInterface::class), $this->createMock(MessageBusInterface::class), $this->createMock(EventDispatcherInterface::class));
+
+        $this->expectException(LogicException::class);
```

Comment:
I would move that line after building the `Email` object, to ensure that we expect the exception during `send`.

The way this `expectException` works in PHPUnit is that it expects the exception to happen after that call, and so the place where we call it is meaningful regarding what the test actually expects (and exception happening before the call would fail the test due to an unexpected error).
Most of the tests in the Symfony testsuite are calling that method at the beginning, but this is because tests were previously written using the `@expectedException` annotation (which is semantically equivalent to calling this `expectException` method at the beginning of the test) and were not refined after migrating.